ast/src/parsed/mod.rs

```diff
@@ -152,21 +152,16 @@ impl<T: FieldElement> ArrayExpression<T> {
 
 impl<T> ArrayExpression<T> {
     /// solve for `*`
-    pub fn solve(&self, degree: DegreeType) -> Option<DegreeType> {
+    pub fn solve(&self, degree: DegreeType) -> DegreeType {
         assert!(degree > 0, "Degree cannot be zero.");
-        // the length of this expression is `a + b*x`
-        let (a, b) = self.len();
-        // it must match `degree`, and we solve for `x`
-        if b == 0 {
-            None
-        } else {
-            assert!(
-                a <= degree,
-                "Array literal is too large ({a}) for degree ({degree})."
-            );
-            assert_eq!((degree - a) % b, 0, "Cannot find a suitable value for `*`");
-            Some((degree - a) / b)
-        }
+        // This finds the constant length of the array expression
+        let (a, _) = self.len();
```

Comment:
I realise `len` was like this before, but can it be changed to returning a single value, or is this used somewhere else?